staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -98,8 +98,38 @@ func ValidateUpdateCustomResourceDefinitionStatus(obj, oldObj *apiextensions.Cus
 	return allErrs
 }
 
+// ValidateCustomResourceDefinitionVersion statically validates. If allowingPerVersionFieldOverride
+// is true, we are updating a CRD that already uses the per-version field override feature, we allow
+// the update operation to continue using the feature even if the CustomResourceWebhookConversion
+// feature gate is set off.
+func ValidateCustomResourceDefinitionVersion(version *apiextensions.CustomResourceDefinitionVersion, fldPath *field.Path, index int, allowingPerVersionFieldOverride bool) field.ErrorList {
+	allErrs := field.ErrorList{}
+	subresourcesInvalid := false
+	if version.Subresources != nil {
+		if !utilfeature.DefaultFeatureGate.Enabled(apiextensionsfeatures.CustomResourceWebhookConversion) && !allowingPerVersionFieldOverride {
+			allErrs = append(allErrs, field.Forbidden(fldPath.Child("subresources"), "disabled by feature-gate CustomResourceWebhookConversion"))
+			subresourcesInvalid = true
+		}
+		if !utilfeature.DefaultFeatureGate.Enabled(apiextensionsfeatures.CustomResourceSubresources) {
+			allErrs = append(allErrs, field.Forbidden(fldPath.Child("subresources"), "disabled by feature-gate CustomResourceSubresources"))
+			subresourcesInvalid = true
+		}
+	}
+	if !subresourcesInvalid {
+		allErrs = append(allErrs, ValidateCustomResourceDefinitionSubresources(version.Subresources, fldPath.Child("subresources"))...)
+	}
+	if version.AdditionalPrinterColumns != nil && !utilfeature.DefaultFeatureGate.Enabled(apiextensionsfeatures.CustomResourceWebhookConversion) && !allowingPerVersionFieldOverride {
+		allErrs = append(allErrs, field.Forbidden(fldPath.Child("columns"), "disabled by feature-gate CustomResourceWebhookConversion"))
+	} else {
+		for i := range version.AdditionalPrinterColumns {
+			allErrs = append(allErrs, ValidateCustomResourceColumnDefinition(&version.AdditionalPrinterColumns[i], fldPath.Child("columns").Index(i))...)
+		}
+	}
+	return allErrs
```

Comment:
we add schema as well. Would expect schema validation here too.

Reply:
Schema validation lives in the multi-version Schema PR #67553.

Reply:
Yeah, but the type change is in here, probably by accident.